config/avahi/avahi.inc

```diff
@@ -203,9 +225,53 @@ function avahi_sync() {
 	}
 	avahi_write_config();
 	// Is package enabled?
-	if (($config['installedpackages']['avahi']['config'][0]['enable']) && file_exists("/usr/local/etc/rc.d/avahi-daemon.sh")) {
+	if ($config['installedpackages']['avahi']['config'][0]['enable']) {
 		start_service("avahi");
 	}
 }
 
+function avahi_upgrade_config() {
+	global $config, $avahi_config;
+
+	$avahi_config = $config['installedpackages']['avahi']['config'][0];
```

Comment:
you should use reference (&=) here, otherwise unset() you call below won't have any effect on $config